packages/x-components/src/components/page-selector.vue

```diff
@@ -0,0 +1,216 @@
+      class="x-button"
+      :class="{
+        'x-button-lead x-cursor-default': page === currentPage,
+        'x-button-ghost': page !== currentPage,
+        'x-cursor-not-allowed': page === '...'
+      }"
+    >
+      <slot name="current-page-button">{{ page }}</slot>
+    </button>
+
+    <button
+      @click="selectPage(currentPage + 1)"
+      class="x-button x-button-lead"
+      :disabled="currentPage === totalPages"
+    >
+      <slot name="next-page-button">Next</slot>
+    </button>
+  </nav>
+</template>
+
+<script lang="ts">
+  import { defineComponent, computed } from 'vue';
+  import { useXBus } from '../composables';
+
+  /**
+   * Component that renders a pagination control with buttons for navigating
+   * between pages. It displays the current page, allows selecting other pages,
+   * and emits events when a page is selected.
+   *
+   * @public
+   */
+  export default defineComponent({
+    name: 'PageSelector',
+    props: {
+      /**
+       * The total number of pages.
+       */
+      totalPages: {
+        type: Number,
+        required: true
+      },
+      /**
+       * The current page number.
+       */
+      currentPage: {
+        type: Number,
+        required: true
+      },
+      /**
+       * The number of pages to show before and after the current page.
+       */
+      range: {
+        type: Number,
+        default: 2
+      }
+    },
+    setup: function (props) {
+      const bus = useXBus();
+
+      const visiblePages = computed<(number | string)[]>(() => {
+        const start = Math.max(props.currentPage - props.range, 1);
+        const end = Math.min(props.currentPage + props.range, props.totalPages);
+        const pages: (number | string)[] = Array.from(
+          { length: end - start + 1 },
+          (_, i) => start + i
+        );
+
+        // Ensure first and last pages are always visible when needed
+        if (start > 1) {
+          pages.unshift(1);
+          if (start > 2) {
+            pages.splice(1, 0, '...');
+          }
+        }
+        if (end < props.totalPages) {
+          if (end < props.totalPages - 1) {
+            pages.push('...');
+          }
+          pages.push(props.totalPages);
+        }
+
+        return pages;
+      });
+
+      /**
+       * Handles the selection of a page.
+       *
+       * @param page - The page to select. Can be a number representing the page number or a string '...' indicating an ellipsis.
+       */
+      const selectPage = (page: number | string): void => {
+        if (page === '...') {
+          return;
+        }
+        if (typeof page === 'number' && page > 0 && page <= props.totalPages) {
+          bus.emit('UserSelectedAPage', page);
+          /**
+           * Emits scroll to top to prevent keeping the position if there is more content
+           * after results, as for example Next Queries Preview.
+           */
+          bus.emit('UserClickedScrollToTop', 'main-scroll');
```

Comment:
You can't hardcode the scroll id here because if you do then this component will only work if it is placed on the main scroll. Otherwise strange behaviours will happen. We might need to think a different way to handle the scroll scenario or make the scroll id configurable.